app/services/key_rotator/hmac_fingerprinter.rb

```diff
@@ -14,7 +14,7 @@ def rotate(user:, pii_attributes: nil, profile: nil)
 
     def rotate_email_fingerprint(user)
       ee = EncryptedAttribute.new_from_decrypted(user.email)
-      user.update_columns(email_fingerprint: ee.fingerprint)
+      UpdateUser.new(user: user, attributes: { email_fingerprint: ee.fingerprint }).call
```

Comment:
`update_columns` is faster, but it doesn't perform any validations or callbacks. Given that we don't have any in the User model (except for `set_default_role`), I think it should be fine to use `update_columns`. I would vote to pick one for consistency.